application-ldapuserimport-api/src/main/java/com/xwiki/ldapuserimport/internal/DefaultLDAPUserImportConfiguration.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.ldapuserimport.internal;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.LocalDocumentReference;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xwiki.ldapuserimport.LDAPUserImportConfiguration;
+
+/**
+ * Default implementation of the {@link LDAPUserImportConfiguration}.
+ *
+ * @version $Id$
+ * @since 1.4
+ */
+@Component
+@Singleton
+public class DefaultLDAPUserImportConfiguration implements LDAPUserImportConfiguration
+{
+    private static final String LDAP_USER_IMPORT = "LDAPUserImport";
+
+    private static final DocumentReference CONFIGURATION_REFERENCE =
+        new DocumentReference("xwiki", LDAP_USER_IMPORT, "WebHome");
```

Comment:
Used `WikiDescriptorManager#getMainWikiId()` as part of 4f96ac8.

Following my mention on https://github.com/xwikisas/application-ldapuserimport/pull/65#discussion_r1177657094, I don't think that we'll be able to use `AbstractDocumentConfigurationSource` at the moment because it's internal.